plone/app/uuid/utils.py

```diff
@@ -31,12 +46,13 @@ def uuidToObject(uuid):
     """Given a UUID, attempt to return a content object. Will return
     None if the UUID can't be found.
     """
-
-    brain = uuidToCatalogBrain(uuid)
-    if brain is None:
-        return None
-
-    return brain.getObject()
+    path = uuidToPhysicalPath(uuid)
+    if not path:
+        return
+    site = getSite()
+    if site is None:
+        return
+    return site.restrictedTraverse(path)
```

Comment:
I think here we need an `unrestrictedTraverse`, otherwise we run into permission problems if i.e. a  path `/a/b/c` with no permission for b but for a and c it will fail. Correct me if I am wrong. We may need a permission check for the resulting object then here.

Reply:
I would do what the `getObject` does, it gets `a/b` unrestricted and the it does a restrictedTraverse to `c`, see:
https://github.com/zopefoundation/Products.ZCatalog/blob/d4fd9fe28d27e8ff43e911025c1258f1e8d50ad5/src/Products/ZCatalog/CatalogBrains.py#L88-L91

Reply:
I switched to the combination of unrestricted and restricted traverse. I catch Unauthorized at the end, and return None then. This should be same as the original, where a catalog query would come up empty when you do not have permission on the object.